### PR TITLE
VIITE-1893 added fix for first check

### PR DESCRIPTION
**Pull request by Campanha**
Added fix for error message in Transfer cases where the ne…w road part was not reserved in the project

### CODE REVIEW

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectValidator.scala [hunk 1]
```diff
@@ -83,8 +83,17 @@ class ProjectValidator {
     }
   }
 
-  def checkNotReserved(number: Long, part: Long, currentProject: Project): Unit = {
-    val project = projectReservedPartDAO.roadPartReservedByProject(number, part, currentProject.id, withoutProjectId = true)
+  def checkReservedPartInProject(number: Long, part: Long, currentProject: Project, linkStatus: LinkStatus): Unit = {
```
Comment (elenaols) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectValidator.scala]: Is it possible to pass LinkStatus as optional parameter and use just one checkReservedPartInOtherProject? And combine ProjectValidationExceptions like:
 if (project.isEmpty) { throw new ProjectValidationException(RoadPartNotReservedInProjectMessage.format(number, part, currentProject.name))
} else {
throw new ProjectValidationException(RoadReservedOtherProjectMessage.format(number, part, currentProject.name)) 
}

Comment (Campanha) [reply digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectValidator.scala]: well ive refactored the DAOs fetch methods a bit but the val project are not the opposite in the functions and the if else will not work n this case

**Comment by Campanha** [comment]
Do not merge: still need to change tests due to new check validations

**Comment by Campanha** [comment]
tests passing at http://livibuild04.vally.local/jenkins/job/viite/job/Run-Tests/9/console

**Pull request merged**